### ISSUE TITLE
FP — the registration pairing fact models only listener-binding teardowns, so two conformant idioms read as `none-seen`

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#124]
Found by running the community packs (which use the new `teardown` guard) over the corpus. Three measured false positives on correct code, all from the same root: [ADR-034](../blob/main/docs/adr/ADR-034-registration-relation.md) §3 pairs a registration with a teardown **by listener binding**, and real teardowns come in three shapes, not one.

## The three shapes

**(a) The returned-handle idiom.** `test-repo/ai-chatbot/components/chat/toolbar.tsx:180`

```tsx
useEffect(() => {
  const unsubscribe = yToLevel.on("change", (latest) => { setCurrentLevel(...); });
  return () => unsubscribe();
}, [yToLevel]);
```

The registration returns its own disposer. The cleanup calls it, and never
mentions the listener. `Pairing::Unpaired` — a claim, and a wrong one.

**(b) A self-removing registration.** `test-repo/excalidraw/packages/excalidraw/data/library.ts:747`

```tsx
window.addEventListener("focus", () => resolve(), { once: true });
```

`{ once: true }` makes the listener remove itself after one dispatch. No
teardown is needed or possible, and the registrar table has no notion of the
options argument.

**(c) A handle-valued teardown.** `setInterval` / `clearInterval(id)`.

```tsx
const id = setInterval(tick, 1000);
return () => clearInterval(id);
```

`clearInterval` takes the *handle*, never the callback, so a binding match can
never succeed. The registrar table lists `clearInterval` as `setInterval`’s
teardown, but §3 compares it against the callback binding and answers
`none-seen`. (The async triage agent had to drop `setInterval` from its rule’s
registrar list to get a usable rule, which is how this surfaced.)

## Why it matters

`Paired` is the only value in the three-valued fact that is a claim, and the
whole design of ADR-034 §3 is that everything unresolvable folds to the may-side
so nothing is *certified* wrongly. That holds. But the may-side is also what a
rule **fires** on, and these three shapes are correct code landing on it — so
`subscribe-with-fresh-listener` and every pack rule built on `teardown` carries
a false-positive class that is not a listener-identity bug at all.

## Fix direction

The teardown column is a list of names today. It needs an *arity/target model*:

- **handle-valued** (`clearInterval`, `clearTimeout`, `cancelAnimationFrame`,
  `cancelIdleCallback`): pair on the binding the registration’s **return value**
  was assigned to, not on its callback argument.
- **listener-valued** (`removeEventListener`, `off`, `removeListener`): today’s
  rule, unchanged.
- **disposer-valued** (the returned-handle idiom): the registration’s return
  binding is *called* in the cleanup. Same fact as the handle case — the
  difference is only whether the cleanup passes it or invokes it.

And `addEventListener`’s third argument: `{ once: true }` (or the boolean-`true`
capture form paired with it) means the registration takes itself back, so the
row should answer `Paired` outright.

All three are the same missing piece — the relation records what a registration
*consumes* and not what it *returns*.

**Comment by rboudrouss** [comment]
Fixed. A teardown identifies its registration by one of three values, and the
relation only knew one of them.

## What changed

`Registrar` gains a `teardown_takes` column — `Listener` or `Handle` — and the
row records the binding the registration's **result** was assigned to.

| shape | example | matched on |
|---|---|---|
| listener-valued | `el.removeEventListener(t, h)`, `off(evt, h)` | the listener binding (unchanged) |
| handle-valued | `clearInterval(id)` | the binding the call returned |
| disposer-valued | `const u = s.subscribe(f); return () => u()` | that binding, *invoked* |

The disposer form is the same fact as the handle form — the difference is only
whether the cleanup passes the handle or calls it — so it is checked whatever
the column says, which is what makes it work for every registrar that returns
something.

Plus the fourth case this issue named: `addEventListener(t, h, { once: true })`
takes itself back, so the row is `Paired` outright. The boolean third argument
is `capture` and does not count.

## Two things deliberately closed

`Paired` is the verdict that *suppresses*, so widening it is the direction that
can produce a false negative:

- the disposer set is a fixed list — `unsubscribe`, `dispose`, `cancel`,
  `close`, `destroy`, `remove`, `off`, `abort` — not "any method called on the
  handle". `s.emit('bye')` is not a teardown, and there is a test that says so.
- the `Unpaired`/`Unknown` split now follows what was comparable: a
  handle-valued registration whose result is never bound, and a listener-valued
  one whose callback is an inline literal, both answer `Unknown` rather than a
  claim.

## The three measured false positives

- `ai-chatbot/components/chat/toolbar.tsx:180` (returned disposer) — **gone**.
- `excalidraw/packages/excalidraw/data/library.ts:747` (`{ once: true }`) — **gone**.
- `setInterval` / `clearInterval(id)` — **paired**, so the async pack rule no
  longer needs `setInterval` dropped from its registrar list to be usable.

One excalidraw finding remains, on `SearchMenu.tsx`, and it is a different
issue: that file imports its **own** four-argument `addEventListener`, so the
name-table match reads argument 1 (the event name) as the callback. That is
wontfix #42's caveat exactly, and the row answers `Unknown` — a may, not a
claim — either way.

Six tests, three of them controls. ADR-034 §3 amended in place; the `teardown`
guard's row in both reference documents updated. Corpus otherwise unchanged, as
expected — no native rule reads the pairing fact and no shipped pack queries it.

**Comment by rboudrouss** [comment]
Fixed in f11a6c7 — both measured false positives gone, corpus otherwise unchanged.
